ezgz.hpp

```diff
@@ -156,7 +156,8 @@ class ByteInput {
 	int refillSome() {
 		if (position > std::ssize(buffer) / 2) {
 			filled -= position;
-			memmove(buffer.data(), &buffer[position], filled);
+            if(filled != 0)
```

Comment:
What is the reason for this? memmove and memcpy are safe to call with size 0 as far as I know?

Reply:
position is equal to buffer.size() when filled is zero

Reply:
Sure, but that's fine unless it's dereferenced, which shouldn't happen for size=0

Reply:
operator[] is not happy about invalid indices, this is not about the memmove.

Reply:
Then you can use buffer.data() + position, which is more symmetrical with source argument anyway.

Reply:
Having a begin iterator or pointer being out of bounds, ie pointing to end, is very common for the case size=0, not just for memcpy, but in general.

Reply:
size of zero is ok for memmove/memcpy but accessing an array/vector via operator[] with an invalid index is not. Feel free to create a separate PR to address this.

Reply:
Yeah, probably will, but tomorrow as it's getting late here.

Reply:
I have done it, no need to bother.